src/5095wrappers/Base5095.sol

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import "../tokens/ERC20Permit.sol";
+import "../interfaces/IERC5095.sol";
+import "../errors/Exception.sol";
+import "../interfaces/IMarketPlace.sol";
+
+import "../lib/Cast.sol";
+import "../lib/Safe.sol";
+
+contract Term5095 is ERC20Permit {
```

Comment:
`Base5095` instead of `Term5095`?

Reply:
Yeah thats the plan at the moment --

Thatll settle out with some base functionality that is shared across all of them, and then each individual one handles the transfer of whatever asset uniquely, and the checks to ensure its of the correct underlying and maturity.

That said theres a duplicate for now for 2 reasons:
1. Term is a bit different than others -- because Term's lending process is asynchronous, there isnt a way to deposit USDC to receive a term position which we would then wrap -- So the `lend` method is empty in its adapter, and the `mint/deposit` methods operate differently as well

So the term one will likely end up being unique, at least relative to the others.

2. The current `Base5095` is largely a placeholder for me to pull logic when creating the other adapters. Once the implementation is settled for the other protocols, I'll remove any unneeded functionality from the Base5095 (stuff overwritten consistently) and then have each individual 5095 inherit like youve suggested 👍